src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -1,5 +1,7 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#nullable enable
```

Comment:
The following members of CompilationOptions also need to be made nullable, at least per the comments:

- ModuleName
- ScriptClassName
- MainTypeName
- CryptoKeyContainer
- MetadataReferenceResolver
- XmlReferenceResolver
- SourceReferenceResolver
- StrongNameProvider

I'd also assume you want to update the constructor parameters to match as well, as well as all the With* methods.

Reply:
> ScriptClassName

The comments make no mention of `null` and constructor enforces it's non-null.

Agree with the rest. 


---
In reply to: [360007738](https://github.com/dotnet/roslyn/pull/40493#discussion_r360007738) [](ancestors = 360007738)